src/PSDMatrices.jl

```diff
@@ -30,8 +30,27 @@ end
 
 # LinearAlgebra overloads
 
-det(M::PSDMatrix) = det(M.R)^2
-logdet(M::PSDMatrix) = 2 * logdet(M.R)
+function det(M::PSDMatrix)
+    if size(M.R, 1) != size(M.R, 2)
+        msg = _errormsg_non_square_factor(M)
+        throw(DimensionMismatch(msg))
+    end
+    return det(M.R)^2
+end
+
+function logdet(M::PSDMatrix)
+    if size(M.R, 1) != size(M.R, 2)
+        msg = _errormsg_non_square_factor(M)
+        throw(DimensionMismatch(msg))
+    end
+    return 2 * logdet(M.R)
+end
+
+_errormsg_non_square_factor(M::PSDMatrix) = (
```

Comment:
Overall this looks fine, but I'm wondering about two details here:
1. did you consider making the function `_non_square_factor_error`, that returns a `DimensionMismatch` with the correct error?
2. or alternatively, have a custom exception? (but I'm not sure how that would be called)
3. and on a different level: Do we need the leading underscore? I'm not sure about Julia conventions here. But since it's not exported, I'd assume that it should also be safe to give it a normal function name. What do you think?

Reply:
1. Yes, but I personally find it slightly more readable if the error is raised on top-level (i.e. on the same level as the return). But this is subjective, and I can change it if you like. 
2. I briefly thought about this, but I don't know which exception that would be... and it might be a bad idea to start construction custom exceptions left and right? Alternatively, one could raise a `MethodError` here, because strictly speaking, the error is not that the dimensions don't match but that PSDMatrices does not know how to do this efficiently. Thoughts?
3. I was wondering the same and found https://stackoverflow.com/questions/53687998/function-name-with-a-leading-underscore. I understood this to be a mild stylistic convention, but it looks familiar to me so I liked it :D Shall I change it?

Reply:
1. no strong preference so let's keep it this way :+1: 
2. now that you mention it, maybe I'd slightly favor a `MethodError` here, but again no real preferences. Regarding a custom error, I again have no idea about conventions, but since I don't know what exactly it should be called I agree that we should rather use existing ones.
3. I also found that stackoverflow question, but nothing else really. If you don't have a preference I'd tend towards not using a leading underscore here to mark the function as internal, since unless exported they all are.

Reply:
But to bring another wild suggestion to the table: What about something like
```julia
function ensure_square_factor(M::PSDMatrix)
    msg = (
        "The requested operation is not available for a PSDMatrix with a non-square factor." *
        "The factor of the received PSDMatrix has dimensions ($(size(M.R,1)), $(size(M.R,2))). " *
        "Try turning the PSDMatrix into a dense matrix first."
    )
    if size(M.R, 1) != size(M.R, 2)
        throw(DimensionMismatch(msg))
    end
end
```
? Maybe we need to check for square square roots (lol) in more places, so that deduplicates things a bit? 

But again, just a random suggestion and if you prefer the current approach that's fine with me :)

Reply:
Nice :D I also wasn't completely happy with `ensure`, `confirm` is better indeed! :+1: